kratos/geometries/nurbs_curve_tessellation.h

```diff
@@ -0,0 +1,227 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Andreas Apostolatos
+//					 Tobias Teschemacher
+//					 Thomas Oberbichler
+//					
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_TESSELLATION_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_TESSELLATION_H_INCLUDED
+
+#include "utilities/math_utils.h"
+#include "geometries/geometry.h"
+#include "geometries/nurbs_curve_geometry.h"
+
+#include "includes/ublas_interface.h"
+#include "containers/array_1d.h"
+
+namespace Kratos {
+
+// template <typename TCurve>
+template <int TWorkingSpaceDimension, class TContainerPointType>
+class NurbsCurveTessellation
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Geometry<typename TContainerPointType::value_type> GeometryType;
+    typedef NurbsCurveGeometry<TWorkingSpaceDimension, TContainerPointType> NurbsCurveGeometryType;
+    typedef typename GeometryType::IndexType IndexType;
+    typedef typename GeometryType::SizeType SizeType;
+
+private:    // static methods
+    static double DistanceToLine(
+        const typename GeometryType::CoordinatesArrayType& Point, 
+        const typename GeometryType::CoordinatesArrayType& LineA,
+        const typename GeometryType::CoordinatesArrayType& LineB
```

Comment:
check the style-guide for variable namings: https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide

Reply:
+1